### DISCUSSION TITLE
❌ Terminal Replacement

### DISCUSSION THREAD

**Discussion by chakrab** [isontheline/pro.webssh.net#431] [Q&A]
Hello,

I found WebSSH recently and now am using it on my Mac, iPad and iPhone (rarely). It is very convenient to access all my SSH in one single place. However, I still have to switch to iTerm2 for my local terminal need. Is there any option on WebSSH to open up a Mac terminal?

Thanks.

**Comment by isontheline** [comment]
Hello @chakrab 👋

Thank you a lot for using WebSSH 🙏

Unfortunately WebSSH rely on Apple' Sandboxing and then can't start a local native terminal on macOS. That's why mashREPL subproject have been started in order to offer a pseudo local terminal.

Yet the only workaround is to add a SSH connection which let you to connect to your macOS through SSH and then you will be able to run local commands.

I will improve mashREPL in the few weeks but it will run in the sandbox though.

Have a nice day ☀️

**Reply by chakrab** [reply]
Thanks for the clarification @isontheline. It is a minor inconvenience - I am sure mashREPL will be able to serve that gap. Thanks again for the nice product 👍.

Have a nice weekend.

**Reply by isontheline** [reply]
Thank you a lot for your support about WebSSH @chakrab 🙏
I let this discussion open in order to track needs around mashREPL.

Nice weekend too ☀️

**Reply by isontheline** [reply]
@chakrab As a workaround you can add a SSH connection to WebSSH that will connect to your own localhost device
